webapp/packages/core-di/src/useService.ts

```diff
@@ -7,23 +7,21 @@
  */
 import { useContext } from 'react';
 
-import { appContext } from './AppContext';
 import type { IServiceConstructor } from './IApp';
-import type { ValueToken } from './InjectionToken';
+import { serviceProviderContext } from './ServiceProviderContext';
 
 export function useService<T>(ctor: IServiceConstructor<T>): T;
 export function useService<T>(ctor: IServiceConstructor<T>, optional: true): T | undefined;
 export function useService<T>(ctor: IServiceConstructor<T>, optional?: boolean): T | undefined {
-  const app = useContext(appContext);
+  const serviceProvider = useContext(serviceProviderContext);
 
-  if (optional && !app.hasServiceByClass(ctor)) {
-    return undefined;
+  if (optional) {
+    try {
+      return serviceProvider.getService(ctor);
+    } catch {
+      return undefined;
+    }
   }
```

Comment:
It's intended to fall to undefined in case we can't initialize the service, for example, when a component is used for displaying errors for the app initialization phase (where the theme service cannot be initialized yet)